src/main/scala/io/moia/protos/teleproto/ReaderImpl.scala

```diff
@@ -134,56 +136,97 @@ object ReaderImpl {
+              }))
+            case TransformParam(from, to) if from <:< weakTypeOf[Option[_]] && (to <:< weakTypeOf[Option[_]]) =>
+              val innerFrom = innerType(c)(from)
+              val innerTo   = innerType(c)(to)
+              Some(assign(withImplicitReader(c)(innerFrom, innerTo) { readerExpr =>
+                q"""$mapping.Reader.optional[$innerFrom, $innerTo](protobuf.${termSymbol.name}, $path)($readerExpr)"""
+              }))
+
+            case TransformParam(from, to) if from <:< weakTypeOf[Seq[_]] && to <:< weakTypeOf[Iterable[_]] =>
+              val innerFrom = innerType(c)(from)
+              val innerTo   = innerType(c)(to)
+              Some(assign(withImplicitReader(c)(innerFrom, innerTo) { readerExpr =>
+                // sequence also needs an implicit collection generator which must be looked up since the implicit for the value reader is passed explicitly
+                val canBuildFrom = q"""implicitly[scala.collection.Factory[$innerTo, $to]]"""
+                q"""$mapping.Reader.sequence[${to.typeConstructor}, $innerFrom, $innerTo](protobuf.${termSymbol.name}, $path)($canBuildFrom, $readerExpr)"""
+              }))
+            case TransformParam(from, to) =>
+              Some(assign(withImplicitReader(c)(from, to) { readerExpr =>
+                q"""$mapping.Reader.transform[$from, $to](protobuf.${termSymbol.name}, $path)($readerExpr)"""
+              }))
+            case ExplicitDefaultParam(expr) =>
+              Some(q"val ${termSymbol.name} = $expr" -> Compatibility.full)
+            case SkippedDefaultParam(_) =>
+              None
+          }
+      }
+    }
+
+    /**
+      * Constructs an expression `PbFailure.combine(monadicParameters..)` for all transformed parameters (could fail).
+      */
+    def forLoop(parameters: List[(TermSymbol, MatchingParam[Type, Tree])], cons: Tree): Tree =
       parameters match {
         case Nil =>
-          (cons, Compatibility.full)
-
+          cons
         case (termSymbol, matchingParam) :: rest =>
-          val (restTransformed, restCompatibility) = forLoop(rest, cons)
+          val restTransformed = forLoop(rest, cons)
+          matchingParam match {
+            case TransformParam(from, to) if from <:< to =>
+              restTransformed
+            case ExplicitDefaultParam(_) =>
+              restTransformed
+            case SkippedDefaultParam(_) =>
+              restTransformed
+            case _ =>
+              q"""${termSymbol.name}.flatMap { case ${termSymbol.name} => $restTransformed }"""
+          }
+      }
 
-          val path = q""""/"+${termSymbol.name.decodedName.toString}"""
+    /**
+      * Constructs an expression `PbFailure.combine(monadicParameters..)` for all transformed parameters (could fail).
+      */
+    def combineErrors(parameters: List[(TermSymbol, MatchingParam[Type, Tree])]): Tree = {
+      val monadicValues =
```

Comment:
Is this flatMap really aggregating monadic values?

Reply:
I call it `convertedValues` now.

Reply:
Needed a specific name for `PbResult[T]` instead of just a `T` that can be copied.